src/main/java/edu/kis/vh/nursery/list/IntLinkedList.java

```diff
@@ -1,40 +1,110 @@
 package edu.kis.vh.nursery.list;
 
+class Node {
+    private int value;
+    private Node prev, next;
+
+    public Node(int i) {
+        setValue(i);
+    }
+
+    public int getValue() {
+        return value;
+    }
+
+    public void setValue(int value) {
+        this.value = value;
+    }
+
+    public Node getPrev() {
+        return prev;
+    }
+
+    public Node getNext() {
+        return next;
+    }
+
+    public void setPrev(Node prev) {
+        this.prev = prev;
+    }
+
+    public void setNext(Node next) {
+        this.next = next;
+    }
+}
+
 public class IntLinkedList {
 
-    Node last;
-    int i;
+    private Node last;
+    private int i;
+    private static final int EMPTY = -1;
 
+    public static int getEMPTY() {
+        return EMPTY;
+    }
+
+    /**
+     * Added value to IntLikedList
+     *
+     * @param i this is the value that will be added to IntLikedList
+     */
     public void push(int i) {
         if (last == null)
             last = new Node(i);
         else {
-            last.next = new Node(i);
-            last.next.prev = last;
-            last = last.next;
+            last.setNext(new Node(i));
+            last.getNext().setPrev(last);
+            last = last.getNext();
         }
     }
 
+    /**
+     * Function of check if IntLikedList is empty
+     *
+     * @return true value if IntLikedList is empty otherwise false
+     */
     public boolean isEmpty() {
         return last == null;
     }
 
-    public boolean isFull() {
-        return false;
-    }
-
+    /**
+     * This function get a top element of IntLikedList without delete it.
+     *
+     * @return top element of IntLikedList
+     */
     public int top() {
         if (isEmpty())
-            return -1;
-        return last.value;
+            return getEMPTY();
+        return last.getValue();
     }
 
+    /**
+     * This function get top element of IntLikedList and delete that element from IntLinkList
+     *
+     * @return top element of IntLikedList
+     */
     public int pop() {
         if (isEmpty())
-            return -1;
-        int ret = last.value;
-        last = last.prev;
+            return getEMPTY();
+        int ret = last.getValue();
+        last = last.getPrev();
         return ret;
     }
 
+    /**
+     * Get the last node  of IntLikedList
+     *
+     * @return last node  of IntLikedList
+     */
+    public Node getLast() {
```

Comment:
Shouldn't be encapsulated (typical private field). This is an exaggeration. Using private fields internally usually do not require encapsulation.